### PR TITLE
Fix alpha bugs and Update DG

### PR DESCRIPTION
**Pull request by vinc3leong**
Include AddCommand class to encapsulate AddSupplierCommand and AddDeliveryCommand
Make DeleteCommand an abstract class that encapsulates DeleteSupplierCommand and DeleteDeliveryCommand
Update MarkSupplierCommand and MarkDeliveryCommand to inherit from MarkCommand 
Include JavaDocs for SortSupplierCommand and a few other constructors 
Improve parsing logic in AddressBookParser
Remove Address Class

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -639,21 +637,97 @@ testers are expected to do more *exploratory* testing.
...
+   1. Test case: `list -s`<br>
+      Expected: All suppliers are shown in the list.
+
+### Mark supplier status
+
+1. Marking a supplier as active/inactive
+
+   1. Prerequisites: List all suppliers using the `list -s` command. Multiple suppliers in the list.
+
+   1. Test case: `mark -s 1 active`<br>
```
Comment (CJianzhi) [inline docs/DeveloperGuide.md:722]: Should we mention that the first supplier must not have active status in the prereq?

Comment (ngjianyi) [inline docs/DeveloperGuide.md:722]: Should there be a negative test case for when marking an active supplier as active again?

**Review by ngjianyi — COMMENTED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -639,21 +637,99 @@ testers are expected to do more *exploratory* testing.
...
+   2. Test case: `find -s n/John`<br>
+      Expected: The suppliers with name containing "John" is shown in the list. Other suppliers are not shown.
+
+   3. Test case: `find -s n/John com/comp`<br>
+      Expected: The suppliers with name containing "John" and company name containing "comp" is shown in the list. Other suppliers are not shown.
+
+   4. Test case: `find -s n/John com/comp pro/rice`<br>
+      Expected: The suppliers with name containing "John", company name containing "comp" and product containing "rice" is shown in the list. Other suppliers are not shown.
+   
+   6. Test case: `find -s n/John com/comp pro/rice pro/bread`<br>
```
Comment (CJianzhi) [inline docs/DeveloperGuide.md:674]: Same for here should be 1.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -639,21 +637,99 @@ testers are expected to do more *exploratory* testing.
...
+#### Adding a supplier
+
+1. Adding a supplier 
+
+   1. Prerequisites: No "John Doe" with company "companyA" and "Jane Doe" with company "companyB" in the list of suppliers.
+
+   1. Test case: `add -s n/John Doe p/98765432 e/johnd@example.com com/companyA t/friends t/owesMoney pro/rice pro/bread`<br>
+     Expected: A new supplier is added to the list.
+
+   2. Test case: `add -s n/Jane Doe p/87654321 e/jane@example.com com/companyB`<br>
```
Comment (CJianzhi) [inline docs/DeveloperGuide.md:651]: i think all the numbering should be 1.

**Review by CJianzhi — APPROVED** [review]
LGTM, some minor changes to be made to developer guide for numbering

**Pull request merged**